libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts

```diff
@@ -310,7 +310,7 @@ export class SelectionParameterSet {
       this._impactedByExceptProject,
       terminal
     )) {
-      args.push('--filter', `...^${project.packageName}`);
+      args.push(`...^${project.packageName}`);
```

Comment:
I think you may have misunderstood my last comment regarding the filter... the issue here is that you won't be parsing these filter arguments correctly. You will instead be checking exact filter arguments instead of the "list of projects that have been installed".

You should pass through a set of package names for each individual package to be installed, not passing the filter spec that PNPM understands. This way you can put this list of projects into the install file instead of the filter specs. When passing this list to PNPM, you would just need to prefix each project name with `--filter <packagename>` to ensure that PNPM installs it.

Reply:
ah I see, let me change that

Reply:
I refactored to use the "getSelectedProjectsAsync" instead of "getPnpmFilterArgumentsAsync" to create the filter criteria to pass the selected projects as options instead of filter criteria